blocks/contact-form/contact-form.js

```diff
@@ -69,27 +75,51 @@ export default function decorate(block) {
       div.removeChild(inputTypeColumn);
     }
   });
+
   // Specify submitContactInfo arguments
   const submitInfo = async (event) => {
     // To prevent the button from scrolling up when clicked
     if (event) {
       event.preventDefault();
     }
 
-    // Make sure no inputs are empty
+    // Select empty inputs
     const emptyInputs = [...formInputs].filter(
-      (input) => input.required && !input.value.trim(),
+      (input) => !input.checkValidity(),
     );
 
-    if (emptyInputs.length > 0) {
+    if (!form.checkValidity()) {
+      // Set className for empty inputs to highlight in red
       emptyInputs.forEach((input) => input.classList.add('required-empty-input'));
+
+      // Set message if info fails to send
+      block.appendChild(failMessage);
+      setTimeout(() => {
+        block.removeChild(failMessage);
+      }, 3000);
       return;
     }
 
     await submitContactInfo(formInputs);
+
+    // Set success message
+    block.appendChild(successMessage);
+    setTimeout(() => {
+      block.removeChild(successMessage);
+    }, 3000);
     form.reset();
   };
 
+  // Remove red border from inputs once they have values by removing the className
+  form.addEventListener('input', (event) => {
+    if (event.target.tagName === 'INPUT') {
+      const input = event.target;
+      if (input.value.trim()) {
```

Comment:
don't forget, a field can be invalid even if it has value, so you might have to change this check